psi4/src/psi4/libfock/jk.cc

```diff
@@ -168,8 +168,9 @@ std::shared_ptr<JK> JK::build_JK(std::shared_ptr<BasisSet> primary, std::shared_
     if (jk_type == "DF") {
 
         // logic for MemDFJK vs DiskDFJK
-        if (do_wK || options["DF_INTS_IO"].has_changed()) {
+        if (do_wK || !auxiliary->has_puream() || options["DF_INTS_IO"].has_changed()) {
```

Comment:
Please add a throw in DFHelper as well, this will likely effect your 3-index transforms and especially if someone builds `MemDFJK` directly.

Reply:
done